tests/bsim/bluetooth/mesh/src/test_access.c

```diff
@@ -522,6 +523,70 @@ static void msgf_publish(void)
 	bt_mesh_model_publish(model);
 }
 
+static void pub_delayable_check(int32_t interval, uint8_t count)
+{
+	int64_t timestamp = k_uptime_get();
+	int err;
+
+	for (size_t j = 0; j < count; j++) {
+		/* Every new publication will release semaphore in the update handler and the time
+		 * between two consecutive publications will be measured.
+		 */
+		err = k_sem_take(&publish_sem, K_SECONDS(20));
+		if (err) {
+			FAIL("Send timed out");
+		}
+
+		int32_t time_delta = k_uptime_delta(&timestamp);
+		int32_t pub_delta = llabs(time_delta - interval);
+
+		LOG_DBG("Send time: %d delta: %d pub_delta: %d", (int32_t)timestamp, time_delta,
+			pub_delta);
+
+		if (j == 0) {
+			ASSERT_IN_RANGE(pub_delta, 0, 310);
+		} else {
+			ASSERT_IN_RANGE(pub_delta, 0, 10);
+		}
+	}
+}
+
+static void recv_delayable_check(int32_t interval, uint8_t count)
+{
+	int64_t timestamp;
+	int err;
+
+	/* The measurement starts by the first received message. */
+	err = k_sem_take(&publish_sem, K_SECONDS(20));
+	if (err) {
+		FAIL("Recv timed out");
+	}
+
+	timestamp = k_uptime_get();
+
+	for (size_t j = 0; j < count; j++) {
+		/* Every new received message will release semaphore in the message handler and
+		 * the time between two consecutive publications will be measured.
+		 */
+		err = k_sem_take(&publish_sem, K_SECONDS(20));
+		if (err) {
+			FAIL("Recv timed out");
+		}
+
+		int32_t time_delta = k_uptime_delta(&timestamp);
+		int32_t pub_delta = llabs(time_delta - interval);
+
+		LOG_DBG("Recv time: %d delta: %d pub_delta: %d", (int32_t)timestamp, time_delta,
+			pub_delta);
+
+		if (j == 0) {
+			ASSERT_IN_RANGE(pub_delta, 0, 300 + RX_JITTER_MAX);
```

Comment:
Will it always work? POSIX PC doesn't give millisecond resolution. Probably it is even worse than 10 msec.
Did you check it with zero randomness?
Will the jitter be much more than the basic value?

Reply:
I don't understand you. What zero randomness, what POSIX PC? We have other test that runs without publication delay.

Reply:
Jitter is:
```
#define RX_JITTER_MAX (10 + CONFIG_BT_MESH_NETWORK_TRANSMIT_COUNT * \
		       (CONFIG_BT_MESH_NETWORK_TRANSMIT_INTERVAL + 10))
```
that means 10 + 2 * (20 + 10) = 70

Does it have sense:
```
ASSERT_IN_RANGE(pub_delta, 0, 30 + RX_JITTER_MAX);
``` 
where jitter is 2 times more than the base value?

What is the reason of such enormously huge `jitter`?

Reply:
Because random can be 500ms and plus jitter. Jitter is used because on of `CONFIG_BT_MESH_NETWORK_TRANSMIT_COUNT` messages can be received.